packages/compass-assistant/src/compass-assistant-provider.tsx

```diff
@@ -272,6 +287,25 @@ export const AssistantProvider: React.FunctionComponent<
         chat.messages = [...chat.messages, contextPrompt];
       }
 
+      const { enableToolCalling } = preferences.getPreferences();
```

Comment:
This is something I'm probably gonna have to check with Sergey when he's back, but the preferences hook isn't working well here - it caches the initial value so it doesn't respond if you toggle the flag from the preferences modal.

This way at least works for now.